tools/distpartitioning/initialize.py

```diff
@@ -476,12 +479,12 @@ def single_dev_init(rank, world_size, func_exec, params, backend="gloo"):
     os.environ["MASTER_PORT"] = '29500'
 
     #create Gloo Process Group
-    dist.init_process_group(backend, rank=rank, world_size=world_size)
+    dist.init_process_group(backend, rank=rank, world_size=world_size, timeout=timedelta(seconds=5*60))
```

Comment:
Can we make `timeout` a tunable parameter?

Reply:
Absolutely. I will add this as a command line argument in the next iteration of commits, when we are addressing newly defined multiple file format.